examples/lsms/lsms.py

```diff
@@ -1,5 +1,200 @@
+from hydragnn.utils.model import print_model
+
+import torch
+import torch.distributed as dist
+
+
+def info(*args, logtype="info", sep=" "):
+    getattr(logging, logtype)(sep.join(map(str, args)))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--preonly",
+        action="store_true",
+        help="preprocess only. Adios or pickle saving and no train",
+    )
+    parser.add_argument("--inputfile", help="input file", type=str, default="lsms.json")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--adios",
+        help="Adios dataset",
+        action="store_const",
+        dest="format",
+        const="adios",
+    )
+    group.add_argument(
+        "--pickle",
+        help="Pickle dataset",
+        action="store_const",
+        dest="format",
+        const="pickle",
+    )
+    parser.set_defaults(format="pickle")
+
+    args = parser.parse_args()
+
+    dirpwd = os.path.dirname(__file__)
+    input_filename = os.path.join(dirpwd, args.inputfile)
+    with open(input_filename, "r") as f:
+        config = json.load(f)
+    hydragnn.utils.setup_log(get_log_name_config(config))
+    ##################################################################################################################
+    # Always initialize for multi-rank training.
+    comm_size, rank = hydragnn.utils.setup_ddp()
+    ##################################################################################################################
+    comm = MPI.COMM_WORLD
+    ## Set up logging
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%%(levelname)s (rank %d): %%(message)s" % (rank),
+        datefmt="%H:%M:%S",
+    )
+
+    os.environ["SERIALIZED_DATA_PATH"] = dirpwd + "/dataset"
+    datasetname = config["Dataset"]["name"]
+    fname_adios = dirpwd + "/dataset/%s.bp" % (datasetname)
+    config["Dataset"]["name"] = "%s_%d" % (datasetname, rank)
+    if args.preonly:
+        for dataset_type, raw_data_path in config["Dataset"]["path"].items():
+            if not os.path.isabs(raw_data_path):
+                raw_data_path = os.path.join(dirpwd, raw_data_path)
+            if not os.path.exists(raw_data_path):
+                raise ValueError("Folder not found: ", raw_data_path)
+            config["Dataset"]["path"][dataset_type] = raw_data_path
+
+        ## each process saves its own data file
+        loader = RawDataLoader(config["Dataset"], dist=True)
+        loader.load_raw_data()
+
+        ## Read total pkl and split (no graph object conversion)
+        hydragnn.preprocess.total_to_train_val_test_pkls(config, isdist=True)
+
+        ## Read each pkl and graph object conversion with max-edge normalization
+        (
+            trainset,
+            valset,
+            testset,
+        ) = hydragnn.preprocess.load_data.load_train_val_test_sets(config, isdist=True)
+
```

Comment:
I think the the if condition at line 69 should me removed. 
Instead, here you should say 

```suggestion
if args.preonly:
    break
```

Reply:
We do not want to remove line 69 because we may want to load existing pickle/adios data with train/test/val splits for reproducibility or continuous training. 
In e6c26ede419d1bad68652df18b7a05793f894c65, I add a new flag `loadexistingsplit` and make it default to do raw data conversion&splitting unless specified. Let me know what you think.